src/platform_impl/linux/x11/mod.rs

```diff
@@ -262,120 +295,166 @@ impl<T: 'static> EventLoop<T> {
     where
         F: FnMut(Event<'_, T>, &RootELW<T>, &mut ControlFlow),
     {
-        let mut control_flow = ControlFlow::default();
-        let mut events = Events::with_capacity(8);
-        let mut cause = StartCause::Init;
-
-        let exit_code = loop {
+        struct IterationResult {
+            deadline: Option<Instant>,
+            timeout: Option<Duration>,
+            wait_start: Instant,
+        }
+        fn single_iteration<T, F>(
```

Comment:
Can this be a closure instead? I think that would make the code shorter and clearer.

Reply:
As far as I know, that's not possible, because the closure would need to hold a mutable reference to things like `self` and `control_flow` but those are also mutably borrowed outside of `single_iteration`, which would produce two mutable borrows at the same time which is not allowed.

Reply:
Ah right, and disjoint closure capture won't save us because it uses methods on `self`.

Then perhaps as a method instead (I know the diff will be larger, but I think it's better that way)?

Reply:
I don't think that would be better. The two are approximately equally good in my opinion.

The method would still have to take everything as a parameter, and it would still have to spell out the type constraints (`F`) on the callback parameter. But that would also force the `IterationResult` struct to be elevated to module level visibility, which isn't needed with the current implementation.

For this reason, I think it would be a waste of time to change `single_iteration` to be a method.

Reply:
Fair enough